### ISSUE TITLE
More Quality Indicators?

### ISSUE THREAD

**Issue by wahln** [e0404/matRad#22]
I'm a big fan of QIs.. maybe include CI, HI, Dx and Vx?

**Comment by wahln** [comment]
In this context, we probably need an additional "prescribedDose" field in the cst-struct for every VOI (or at least targets), since the optimization constraint is not necessarily the same as the prescribed dose.

**Comment by markbangert** [comment]
Niklas, I figure this has all been implemented by you or are there any additional ideas pending?

**Comment by wahln** [comment]
The QIs have been implemented by me, there is however still some stuff that could be done in this context

- atm there is still no designated prescribed dose in matRad for a target.. that means i had to take the objective reference dose for some calculations (Confirmity index etc). This is not very nice, and maybe we should incorporate a prescribed dose somehow?
- I don't like that matRad_calcDVH is doing all the work of calculating DVHs, calling matRad_calcQualityIndicators, and plotting. I would recommend (and have already partially implemented this for myself) that matRad_calcDVHs only calculates the DVHs and returns them, and to outsource visualization and function calling into a wrapper function like matRad_planAnalysis that calls the QI and DVH functions and shows their result. Would be more modular...

**Comment by markbangert** [comment]
OK. I agree that these are valid points.

Regarding #2: If you have partially implemented this, maybe you can prepare
a pull request in the future (low priority - maybe the developments align
with some later work)?
Regarding #1: Let's discuss offline!

**Comment by wahln** [comment]
I generalized this issue to the problem of having a better management of results at all (resultGUI is a weird struct). results should be stored in a more descriptive way.
At the same time, we could enable a system that allows for holding multiple plans in the workspace with the associated results (dose cube, dvhs, quality indicators).

**Comment by becker89** [comment]
Yes agree that the result management could be improved. What structure do you have in mind ?
